[FR005]Boost Beast integration.

wiatrM opened the pull request:
Let's merge as boost beast can be completed. We will do listeners and REST endpoint (handler) as a separate feature.

Please review all the changes.

---
Todo:
- [x] Listener
- [x] Http connection
- [x] Server class
- [x] Connection timeout (read)
- [x] Connection timeout (write)

wiatrM commented on kazesl/test/server/impl/TestServer.cpp:
```diff
@@ -0,0 +1,95 @@
+        serverPort.back().port = 0;
+
+        auto ep = server->endpoints(serverPort);
+        ASSERT_EQ(ep.size(), 1);
```
Yeah, I know. More test will be written when REST endpoint and true listener will be ready.

lapek commented:
Why a external/json/json.hpp is added to repository? Try tu add it once again as a submodule - link to repo is in that file' header.

wiatrM commented:
Reworked json package as @lapek described. Added connection timeout feature (for now only read). Added WIP flag to complete write timeout on connection.

wiatrM commented:
@lapek could we mege?
